### DISCUSSION TITLE
Logging to Memory

### DISCUSSION THREAD

**Discussion by zethon** [gabime/spdlog#2888] [Q&A]
Is there a memory logger that would allow me to progmatically iterate the logs written to it? 

What I'd like to do is create the memory logger (almost) immediately after the app starts, then once the app has loaded its config (and hence it's logging settings) I'd like to write the previously written logs to a file/whatever per the configuration, and then remove the memory logger.

Any thoughts on how I could do this? Thank you!

**Comment by tt4g** [comment]
You can use ringbuffer_sink.
Or create your own sink: https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink
